### PR TITLE
Changes for CSI 1.0

### PR DESCRIPTION
**Pull request by dimm0**

### CODE REVIEW

File: deploy/kubernetes/csi-cvmfsplugin.yaml [hunk 1]
```diff
@@ -11,31 +12,30 @@ spec:
...
-            - "--v=5"
-            - "--csi-address=$(ADDRESS)"
-          env:
-            - name: ADDRESS
-              value: /var/lib/kubelet/plugins/csi-cvmfsplugin/csi.sock
-            - name: KUBE_NODE_NAME
-              valueFrom:
-                fieldRef:
-                  fieldPath: spec.nodeName
+            - "--csi-address=/csi/csi.sock"
```
Comment (rochaporto) [inline deploy/kubernetes/csi-cvmfsplugin.yaml]: I've been testing this. Are /csi and /registration requirements now? I found something in the registrar code, but no docs.

File: deploy/kubernetes/csi-cvmfsplugin.yaml [hunk 2]
```diff
@@ -11,31 +12,30 @@ spec:
...
+            - name: registration-dir
+              mountPath: /registration
         - name: csi-cvmfsplugin
           securityContext:
             privileged: true
             capabilities:
               add: ["SYS_ADMIN"]
             allowPrivilegeEscalation: true
-          image: csi-cvmfsplugin:v0.3.0
+          image: gitlab-registry.nautilus.optiputer.net/prp/cvmfs-csi:latest
```
Comment (rochaporto) [inline deploy/kubernetes/csi-cvmfsplugin.yaml]: Maybe we keep the base image name here? And latest in the default is maybe not ideal, keeping a tag there are no surprises for default deployments.

File: deploy/kubernetes/csi-cvmfsplugin.yaml [hunk 3]
```diff
@@ -48,7 +48,7 @@ spec:
                   fieldPath: spec.nodeName
             - name: CSI_ENDPOINT
               value: unix://var/lib/kubelet/plugins/csi-cvmfsplugin/csi.sock
-          imagePullPolicy: "IfNotPresent"
+          imagePullPolicy: "Always"
```
Comment (rochaporto) [inline deploy/kubernetes/csi-cvmfsplugin.yaml]: Do we need this?

File: deploy/kubernetes/csi-cvmfsplugin.yaml [hunk 4]
```diff
@@ -62,19 +62,22 @@ spec:
               readOnly: true
             - name: host-dev
               mountPath: /dev
+            - name: cvmfs-config
```
Comment (rochaporto) [inline deploy/kubernetes/csi-cvmfsplugin.yaml]: If you feel like it, it would be nice to move this to a separate PR. Otherwise leave it, that's ok too.

File: deploy/kubernetes/csi-nodeplugin-rbac.yaml [hunk 5]
```diff
@@ -1,13 +1,14 @@
 apiVersion: v1
 kind: ServiceAccount
 metadata:
-  name: csi-nodeplugin
+  name: cvmfs-nodeplugin
+  namespace: cvmfs
```
Comment (rochaporto) [inline deploy/kubernetes/csi-nodeplugin-rbac.yaml]: As above.

File: deploy/kubernetes/namespace.yaml [hunk 6]
```diff
@@ -0,0 +1,4 @@
+apiVersion: v1
```
Comment (rochaporto) [inline deploy/kubernetes/namespace.yaml]: Not sure we need this, as above.

File: example/storageclass.yaml [hunk 7]
```diff
@@ -2,6 +2,7 @@ apiVersion: storage.k8s.io/v1
 kind: StorageClass
 metadata:
   name: csi-cvmfs
+  namespace: cvmfs
```
Comment (rochaporto) [inline example/storageclass.yaml]: As above.

**Comment by rochaporto** [comment]
Also.. maybe we can rebase into one single commit 'update to csi 1.0'? I've tried it here to test:
https://github.com/cernops/cvmfs-csi/tree/pr-rebase

**Comment by sfiligoi** [comment]
The name "cvmfs" for the namespace is not strictly needed...
It is just a convention we used.
It may be removed from the default example yamls.

File: deploy/kubernetes/csi-cvmfsplugin-provisioner.yaml [hunk 8]
```diff
@@ -2,6 +2,7 @@ kind: Service
 apiVersion: v1
 metadata:
   name: csi-cvmfsplugin-provisioner
+  namespace: cvmfs
```
[the remark this answers is not in the document]
Comment (sfiligoi) [reply deploy/kubernetes/csi-cvmfsplugin-provisioner.yaml]: Not strictlly needed, see below.

File: deploy/kubernetes/csi-cvmfsplugin.yaml [hunk 9]
```diff
@@ -2,6 +2,7 @@ kind: DaemonSet
 apiVersion: apps/v1beta2
 metadata:
   name: csi-cvmfsplugin
+  namespace: cvmfs
```
[the remark this answers is not in the document]
Comment (sfiligoi) [reply deploy/kubernetes/csi-cvmfsplugin.yaml]: Not strictly needed, see comment below.

File: deploy/kubernetes/csi-cvmfsplugin.yaml (hunk 3 above)
Comment (sfiligoi) [reply deploy/kubernetes/csi-cvmfsplugin.yaml]: Probably not, just our preference.

File: deploy/kubernetes/csi-nodeplugin-rbac.yaml (hunk 5 above)
Comment (sfiligoi) [reply deploy/kubernetes/csi-nodeplugin-rbac.yaml]: The namespace is not strictly needed.
But the name should contain cvmfs in the it, to avoid conflicts.

File: example/storageclass.yaml (hunk 7 above)
Comment (sfiligoi) [reply example/storageclass.yaml]: Not strictly needed, see comment below.

File: deploy/kubernetes/csi-cvmfsplugin.yaml (hunk 2 above)
Comment (sfiligoi) [reply deploy/kubernetes/csi-cvmfsplugin.yaml]: Yes, the existing image is just what we used internally. Should not be the one referenced from the upstream.

File: deploy/kubernetes/csi-cvmfsplugin.yaml (hunk 4 above)
Comment (sfiligoi) [reply deploy/kubernetes/csi-cvmfsplugin.yaml]: I agree it is not strictly needed for moving to v1.

File: deploy/kubernetes/namespace.yaml (hunk 6 above)
Comment (sfiligoi) [reply deploy/kubernetes/namespace.yaml]: Having a dedicated namespace is good practice.
Picking "cvmfs" as the namespace name is an admin choice, although it does seem an obvious one.

File: deploy/kubernetes/csi-cvmfsplugin-attacher.yaml [hunk 10]
```diff
@@ -2,6 +2,7 @@ kind: Service
 apiVersion: v1
 metadata:
   name: csi-cvmfsplugin-attacher
+  namespace: cvmfs
```
[the remark this answers is not in the document]
Comment (sfiligoi) [reply deploy/kubernetes/csi-cvmfsplugin-attacher.yaml]: Not strictly needed, see comment below.

File: deploy/kubernetes/namespace.yaml (hunk 6 above)
Comment (rochaporto) [reply deploy/kubernetes/namespace.yaml]: We're running all the system stuff in kube-system, including cvmfs.

Comment (sfiligoi) [reply deploy/kubernetes/namespace.yaml]: As I said, not essential. 
No objections from my part to remove the use of a dedicated namespace from the CERN git repo.

**Comment by rochaporto** [comment]
@dimm0 closing as it's been merged already, let me know if i missed something.
